src/metax_api/models/catalog_record.py

```diff
@@ -1074,6 +1081,8 @@ def _pre_create_operations(self, pid_type=None):
                 'Catalog %s is a legacy catalog - not generating pid'
                 % self.data_catalog.catalog_json['identifier']
             )
+        elif self.state_is_draft(self.request):
+            self.state = self.STATE_DRAFT
```

Comment:
When saving as draft, the current ``self.identifier`` should be set as the value of ``self.research_dataset['preferred_identifier']``. The value would then be overwritten by the actual value when the dataset is published. The lines 1101-1102 should be moved somewhere above this current if-else block, so that ``self.identifier`` is available for use here.